middleware/logging.go

```diff
@@ -107,29 +113,33 @@ func (l Log) Wrap(next http.Handler) http.Handler {
 
 		case 100 <= statusCode && statusCode < 500 || statusCode == http.StatusBadGateway || statusCode == http.StatusServiceUnavailable:
 			if l.LogRequestAtInfoLevel {
-				requestLog.Infof("%s %s (%d) %s", r.Method, uri, statusCode, time.Since(begin))
-
 				if l.LogRequestHeaders && headers != nil {
-					requestLog.Infof("ws: %v; %s", IsWSHandshakeRequest(r), string(headers))
+					lazySprintf := dskit_log.NewGokitLazySprintf("%s %s (%d) %s ws: %v; %s", []interface{}{r.Method, uri, statusCode, time.Since(begin), IsWSHandshakeRequest(r), string(headers)})
+					level.Info(requestLog).Log("msg", lazySprintf)
+				} else {
+					lazySprintf := dskit_log.NewGokitLazySprintf("%s %s (%d) %s", []interface{}{r.Method, uri, statusCode, time.Since(begin)})
+					level.Info(requestLog).Log("msg", lazySprintf)
+				}
+			} else {
+				if l.LogRequestHeaders && headers != nil {
+					lazySprintf := dskit_log.NewGokitLazySprintf("%s %s (%d) %s ws: %v; %s", []interface{}{r.Method, uri, statusCode, time.Since(begin), IsWSHandshakeRequest(r), string(headers)})
+					level.Debug(requestLog).Log("msg", lazySprintf)
+				} else {
+					lazySprintf := dskit_log.NewGokitLazySprintf("%s %s (%d) %s", []interface{}{r.Method, uri, statusCode, time.Since(begin)})
+					level.Debug(requestLog).Log("msg", lazySprintf)
 				}
-				return
-			}
-
-			requestLog.Debugf("%s %s (%d) %s", r.Method, uri, statusCode, time.Since(begin))
-			if l.LogRequestHeaders && headers != nil {
-				requestLog.Debugf("ws: %v; %s", IsWSHandshakeRequest(r), string(headers))
 			}
 		default:
-			requestLog.Warnf("%s %s (%d) %s Response: %q ws: %v; %s",
-				r.Method, uri, statusCode, time.Since(begin), buf.Bytes(), IsWSHandshakeRequest(r), headers)
+			lazySprintf := dskit_log.NewGokitLazySprintf("%s %s (%d) %s Response: %q ws: %v; %s", []interface{}{r.Method, uri, statusCode, time.Since(begin), buf.Bytes(), IsWSHandshakeRequest(r), headers})
+			level.Warn(requestLog).Log("msg", lazySprintf)
 		}
 	})
 }
 
 // Logging middleware logs each HTTP request method, path, response code and
 // duration for all HTTP requests.
 var Logging = Log{
```

Comment:
Who uses it? I can't see any usage a part from tests. Can you check if Mimir, Loki, Tempo or Pyroscope uses it? Otherwise we should remove it.

Reply:
I have checked `Mimir`, `Loki`, `Tempo` and `Pyroscope` and none of them uses this method.
The only usage is in a test in `dskit`.
Therefore, I will delete it and use the global logger in the tests.